x-pack/plugins/security_solution/public/timelines/components/graph_overlay/index.tsx

```diff
@@ -0,0 +1,150 @@
+} from '../../../timelines/store/timeline/actions';
+
+import * as i18n from './translations';
+
+const OverlayContainer = styled.div<{ bodyHeight?: number }>`
+  height: ${({ bodyHeight }) => (bodyHeight ? `${bodyHeight}px` : 'auto')};
+  width: 100%;
+`;
+
+interface OwnProps {
+  bodyHeight?: number;
+  graphEventId?: string;
+  timelineId: string;
+}
+
+const GraphOverlayComponent = ({
+  bodyHeight,
+  graphEventId,
+  status,
+  timelineId,
+  title,
+}: OwnProps & PropsFromRedux) => {
+  const dispatch = useDispatch();
+  const { navigateToApp } = useKibana().services.application;
+  const onCloseOverlay = useCallback(() => {
+    dispatch(updateTimelineGraphEventId({ id: timelineId, graphEventId: '' }));
+  }, [dispatch, timelineId]);
+  const [showCaseModal, setShowCaseModal] = useState<boolean>(false);
+  const onOpenCaseModal = useCallback(() => setShowCaseModal(true), []);
+  const onCloseCaseModal = useCallback(() => setShowCaseModal(false), [setShowCaseModal]);
+  const currentTimeline = useSelector((state: State) =>
+    timelineSelectors.selectTimeline(state, timelineId)
+  );
+  const onRowClick = useCallback(
+    (id: string) => {
+      onCloseCaseModal();
+
+      dispatch(
+        setInsertTimeline({
+          graphEventId,
+          timelineId,
+          timelineSavedObjectId: currentTimeline.savedObjectId,
+          timelineTitle: title.length > 0 ? title : UNTITLED_TIMELINE,
+        })
+      );
+
+      navigateToApp(`${APP_ID}:${SecurityPageName.case}`, {
+        path: getCaseDetailsUrl({ id }),
+      });
+    },
+    [currentTimeline, dispatch, graphEventId, navigateToApp, onCloseCaseModal, timelineId, title]
+  );
+
+  return (
+    <OverlayContainer bodyHeight={bodyHeight}>
+      <EuiHorizontalRule margin="none" />
+      <EuiFlexGroup gutterSize="none" justifyContent="spaceBetween">
+        <EuiFlexItem grow={false}>
+          <EuiButtonEmpty onClick={onCloseOverlay} size="xs">
+            {i18n.BACK_TO_EVENTS}
+          </EuiButtonEmpty>
+        </EuiFlexItem>
+        <EuiFlexItem grow={false}>
+          <EuiFlexGroup gutterSize="none">
+            <EuiFlexItem grow={false}>
+              <NewCase
+                compact={true}
+                graphEventId={graphEventId}
+                onClosePopover={noop}
+                timelineId={timelineId}
+                timelineTitle={title}
+                timelineStatus={status}
+              />
+            </EuiFlexItem>
+            <EuiFlexItem grow={false}>
+              <ExistingCase
+                compact={true}
+                onClosePopover={noop}
+                onOpenCaseModal={onOpenCaseModal}
+                timelineStatus={status}
+              />
+            </EuiFlexItem>
+          </EuiFlexGroup>
+        </EuiFlexItem>
+      </EuiFlexGroup>
+
+      <EuiHorizontalRule margin="none" />
+      <EuiTitle>
+        <>{`Resolver graph for event _id ${graphEventId}`}</>
+      </EuiTitle>
+      <AllCasesModal
```

Comment:
I am wondering if it will make sense to make `AllCasesModal` smarter so we can avoid duplication of code to manage to show the modal. By creating a hook `useAllCasesModal` who will return all the modal, showCaseModal, onCloseModal and onRowClick.

Reply:
> I am wondering if it will make sense to make `AllCasesModal` smarter so we can avoid duplication of code to manage to show the modal. By creating a hook `useAllCasesModal` who will return all the modal, showCaseModal, onCloseModal and onRowClick.

CC'ing @stephmilovic (for a future PR)

Reply:
ok so let's create an issue so we do not forget about it

Reply:
here we go https://github.com/elastic/kibana/issues/70115